components/market/ticket/TicketPurchaseModal.tsx

```diff
@@ -83,6 +90,77 @@ function TicketPurchaseModal({ open, setOpen, ticket, event }: P) {
     }
   }
 
+  const handlePrimaryTicketPurchase = (
+    contract: any,
+    address: `0x${string}`,
+    price: ethers.BigNumber
+  ) => {
+    contract.methods
+      .buyTicket(
+        `${process.env.NEXT_PUBLIC_WARDEN_API_URL}/ticket/metadata?path=${event._id}/${ticket.ticketMetadata[0].name}`,
+        TicketTypeKey[ticket.ticketType]
```

Comment:
maybe use ?. in case of undefined

Reply:
which one you mean? `contract` ?

If that's the case then dont' worry because it must be checked before being passed into this function already